sonic_platform_base/sonic_ssd/ssd_generic.py

```diff
@@ -0,0 +1,157 @@
+#
+# ssd_generic.py
+#
+# Generic implementation of the SSD health API
+# SSD models supported:
+#  - InnoDisk
+#  - StorFly
+#  - Virtium
+
+try:
+    import exceptions              # Python 2
+except ImportError:
+    import builtins as exceptions  # Python 3
+try:
+    import re
+    import subprocess
+    from .ssd_base import SsdBase
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+SMARTCTL = "smartctl {} -a"
+INNODISK = "iSmart -d {}"
+VIRTIUM  = "SmartCmd -m {}"
+
+NOT_AVAILABLE = "N/A"
+
+
+class SsdUtil(SsdBase):
+    """
+    Generic implementation of the SSD health API
+    """
+    model = NOT_AVAILABLE
+    serial = NOT_AVAILABLE
+    firmware = NOT_AVAILABLE
+    temperature = NOT_AVAILABLE
+    health = NOT_AVAILABLE
+    ssd_info = NOT_AVAILABLE
+    vendor_ssd_info = NOT_AVAILABLE
+
+    def __init__(self, diskdev):
+        self.vendor_ssd_utility = {
+            "Generic"  : { "utility" : SMARTCTL, "parser" : self.parse_generic_ssd_info },
+            "InnoDisk" : { "utility" : INNODISK, "parser" : self.parse_innodisk_info },
+            "M.2"      : { "utility" : INNODISK, "parser" : self.parse_innodisk_info },
+            "StorFly"  : { "utility" : VIRTIUM,  "parser" : self.parse_virtium_info },
+            "Virtium"  : { "utility" : VIRTIUM,  "parser" : self.parse_virtium_info }
+        }
+
+        self.dev = diskdev
+        # Generic part
+        self.fetch_generic_ssd_info(diskdev)
+        self.parse_generic_ssd_info()
+
+        # Known vendor part
+        if self.model:
+            model_short = self.model.split()[0]
+            if self.vendor_ssd_utility.has_key(model_short):
+                self.fetch_vendor_ssd_info(diskdev, model_short)
+                self.parse_vendor_ssd_info(model_short)
+            else:
+                # No handler registered for this disk model
+                pass
+        else:
+            # Failed to get disk model
```

Comment:
suggest to add an error message or at least a warning to indicate info failed. otherwise the pass does not indicate it

Reply:
Updated to return "Unknown" as disk model name. So, the cli output will be
```
root@r-anaconda-10:/home/admin# show platform ssdhealth
Device Model : Unknown
Health       : N/A%
Temperature  : N/AC

```

Reply:
I think that if we define N/A the % or the C should not be printed. Can you change it?

Reply:
Fixed in CLI utility https://github.com/Azure/sonic-utilities/pull/587#issuecomment-531804187 (utility adds units sign)